File: TP/TP2/Code/5ABC.cpp
/* 1. Procédure -> Afficher N fois la caractère C sur une ligne */
/* 2. Procédure -> Afficher deux caractère C, entre eux il y a des espaces */
/* 3. Procédure -> Afficher le contour d'un carré, en utilisant ^^ */
#include <iostream>

/* Procédure 1 */

void lignePleine(int n, char c)
{
  for (int i = 0; i < n; ++i)
    std::cout << c << std::flush;
}

/* Procédure 2 */

void ligneCreuse(int n, char c)
{
  for (int i = 0; i < n; ++i)
    if (i == 0 || i == n - 1)
      std::cout << c << std::flush;
    else
      std::cout << " " << std::flush;
}

/* Procédure 3 */

void afficheCarre(int n, char c)
{
  lignePleine(n, c);
  std::cout << std::endl;
  
  for (int i = 0; i < n - 2; ++i) {
    ligneCreuse(n, c);
    std::cout << std::endl;
  }
  
  lignePleine(n, c);
}

/* Utilisation */

int main()
{
  afficheCarre(10, '*');
  
  std::cout << std::endl;
  return 0;
}

File: TP/TP2/Code/Python/5ABC.py
#!/usr/bin/python
# 1. Procédure -> Afficher N fois la caractère C sur une ligne
# 2. Procédure -> Afficher deux caractère C, entre eux il y a des espaces
# 3. Procédure -> Afficher le contour d'un carré, en utilisant ^^


def ligne_pleine(val, char):
    for i in range(val):
        print(char, end='')


def ligne_creuse(val, char):
    for i in range(val):
        if i == 0 or i == val - 1:
            print(char, end='')
        else:
            print(' ', end='')


def afficher_carre(val, char):
    ligne_pleine(val, char)
    print('')

    for i in range(val - 2):
        ligne_creuse(val, char)
        print('')

    ligne_pleine(val, char)


afficher_carre(10, '*')  # Debug
